spring-integration-core/src/test/java/org/springframework/integration/gateway/AsyncGatewayTests.java

```diff
@@ -204,6 +210,59 @@ public void futureWithWildcardReturned() throws Exception {
 		assertThat(result).isEqualTo("foobar");
 	}
 
+	@Test
+	public void futureVoid() throws Exception {
+		GatewayProxyFactoryBean proxyFactory = new GatewayProxyFactoryBean(TestEchoService.class);
+		proxyFactory.setDefaultRequestChannel(new NullChannel());
+		proxyFactory.setBeanName("testGateway");
+		proxyFactory.setBeanFactory(mock(BeanFactory.class));
+		proxyFactory.afterPropertiesSet();
+		TestEchoService service = (TestEchoService) proxyFactory.getObject();
+		Future<Void> f = service.asyncSendAndForget("test1");
+		Object result = f.get(10, TimeUnit.SECONDS);
+		assertThat(result).isNull();
+
+		new DirectFieldAccessor(proxyFactory).setPropertyValue("initialized", false);
+		proxyFactory.setDefaultRequestChannel((message, timeout) -> {
+			throw new MessageDispatchingException(message, "intentional dispatcher error");
+		});
+		proxyFactory.afterPropertiesSet();
+
+		Future<Void> futureError = service.asyncSendAndForget("test2");
+		assertThatExceptionOfType(ExecutionException.class)
+				.isThrownBy(() -> futureError.get(10, TimeUnit.SECONDS))
+				.withCauseInstanceOf(MessageDispatchingException.class)
+				.withMessageContaining("intentional dispatcher error");
+	}
+
+	@Test
+	public void futureVoidReply() throws Exception {
+		QueueChannel requestChannel = new QueueChannel();
+		CountDownLatch readyForReplyLatch = new CountDownLatch(1);
+		new Thread(() -> {
+			try {
+				Message<?> input = requestChannel.receive();
+				CompletableFuture<Void> reply = new CompletableFuture<>();
+				((MessageChannel) input.getHeaders().getReplyChannel()).send(new GenericMessage<>(reply));
+				readyForReplyLatch.await(10, TimeUnit.SECONDS);
+				reply.complete(null);
+			}
+			catch (InterruptedException ex) {
+				ReflectionUtils.rethrowRuntimeException(ex);
```

Comment:
`Thread.currentThread.interrupt()` before rethrow.

Reply:
This is a unit test. Why do we need to care about that?

Reply:
It's just always good practice to do that - e.g. if JUnit does a wait anywhere later on this thread, it will block because the interrupted flag has been reset.

Probably not an issue, but might be.

Reply:
I didn't notice it was a `new Thread()` so less important, I agree, but my eye is always drawn to these; years ago, I got into the practice of automatically adding it; I think it's a good practice so you don't miss one in an important place.